packages/fluentui-rc/src/SelectWidget/SelectWidget.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  ariaDescribedByIds,
+  enumOptionsIndexForValue,
+  enumOptionsValueForIndex,
+  FormContextType,
+  labelValue,
+  RJSFSchema,
+  StrictRJSFSchema,
+  WidgetProps,
+} from '@rjsf/utils';
+import { Dropdown, Field, Option } from '@fluentui/react-components';
+import { OptionOnSelectData } from '@fluentui/react-combobox';
+
+function getValue(data: OptionOnSelectData, multiple: boolean) {
+  if (multiple) {
+    return data.selectedOptions;
+  }
+  return data.selectedOptions[0];
+}
+
+/** The `SelectWidget` is a widget for rendering dropdowns.
+ *  It is typically used with string properties constrained with enum options.
+ *
+ * @param props - The `WidgetProps` for this component
+ */
+function SelectWidget<T = any, S extends StrictRJSFSchema = RJSFSchema, F extends FormContextType = any>({
+  id,
+  options,
+  label,
+  hideLabel,
+  value,
+  required,
+  disabled,
+  readonly,
+  multiple = false,
+  autofocus = false,
+  rawErrors = [],
+  onChange,
+  onBlur,
+  onFocus,
+}: WidgetProps<T, S, F>) {
+  const { enumOptions, enumDisabled, emptyValue: optEmptyVal } = options;
+
+  const selectedIndexes = enumOptionsIndexForValue<S>(value, enumOptions, multiple);
+  let selectedIndexesAsArray: string[] = [];
+
+  if (typeof selectedIndexes === 'string') {
+    selectedIndexesAsArray = [selectedIndexes];
+  } else if (Array.isArray(selectedIndexes)) {
+    selectedIndexesAsArray = selectedIndexes.map((index) => String(index));
+  }
+
+  const dropdownValue = selectedIndexesAsArray
+    .map((index) => (enumOptions ? enumOptions[Number(index)].label : undefined))
+    .join(', ');
+
+  const _onBlur = () => onBlur(id, selectedIndexes);
+  const _onFocus = () => onFocus(id, selectedIndexes);
+  const _onChange = (_: any, data: OptionOnSelectData) => {
+    const newValue = getValue(data, multiple);
+    return onChange(enumOptionsValueForIndex<S>(newValue, enumOptions, optEmptyVal));
+  };
+
+  return (
+    <Field
+      // @ts-expect-error todo: TS2322: Type 'false' is not assignable to type 'WithSlotShorthandValue<WithSlotRenderFunction<Omit<ComponentProps<LabelSlots>, "required"> & ...
+      // See https://github.com/rjsf-team/react-jsonschema-form/issues/3946
```

Comment:
Can you do me a favor and just try the change to `labelValue()` in the `utils` that I suggested to see if your issue goes away? If so, I'm totally fine with you adding that to this PR

Reply:
Yeah I did, but it didn't work unfortunately. I mentioned that at your first suggestion. The issue is that the return type can be false or a React component, which the Fluent UI label doesn't support. I'm not sure yet how to fix that in a nice way...

Reply:
Is the label you are getting not a `string` or undefined? The override type should return just a string when the `label` a string.